File: LeetCode/Medium/0334-increasing-triplet-subsequence/0334-increasing-triplet-subsequence.java
class Solution {
    public boolean increasingTriplet(int[] nums) {
        int first = Integer.MAX_VALUE;
        int second = Integer.MAX_VALUE;
        for (int i = 0; i < nums.length; ++i){
            if (first > nums[i]) first = nums[i];
            else if (first < nums[i] && second > nums[i]) second = nums[i];
            else if (nums[i] > second) return true;
        }
        return false;
    }
}

File: LeetCode/Medium/0334-increasing-triplet-subsequence/0334-increasing-triplet-subsequence.py
class Solution:
    def increasingTriplet(self, nums: List[int]) -> bool:
        first, second = float("inf"), float("inf")
        for v in nums:
            if v <= first:
                first = v
            elif v <= second:
                second = v
            else:
                return True
        return False